src/dimension_slice.c

```diff
@@ -983,6 +1025,216 @@ ts_dimension_slice_insert_multi(DimensionSlice **slices, Size num_slices)
 	return n;
 }
 
+/*
+ * Insert correlated constraint dimension slices into the catalog.
+ *
+ * Create entries with MINVALUE/MAXVALUE as ranges for correlated
+ * constraints and insert these entries. This allows for all the
+ * chunks to be picked up when queries use correlated columns in
+ * WHERE clauses.
+ *
+ * Also, add these to the chunk's constraint lists so that they
+ * eventually get added into the metadate in this same transaction
+ *
+ * We will asynchronously convert the entries into proper ranges
+ * later
+ *
+ * Returns the number of slices inserted.
+ */
+int
+ts_correlated_constraints_dimension_slice_insert(const Hypertable *ht, Chunk *chunk)
+{
+	Catalog *catalog;
+	Relation rel;
+	Size i = 0;
+	Hyperspace *hs = ht->correlated_space;
+
+	if (hs == NULL)
+		return i;
+
+	catalog = ts_catalog_get();
+	rel = table_open(catalog_get_table_id(catalog, DIMENSION_SLICE), RowExclusiveLock);
+
+	for (i = 0; i < hs->num_dimensions; i++)
+	{
+		Dimension *dim = &hs->dimensions[i];
+		DimensionSlice *slice;
+		DimensionVec *vec;
+
+		/* Check if an entry already exists first */
+		vec = ts_dimension_slice_equal_scan(dim->fd.id,
+											DIMENSION_SLICE_MINVALUE,
+											DIMENSION_SLICE_MAXVALUE);
+		/* Add a new entry if none exists */
+		if (vec->num_slices == 0)
+		{
+			slice = ts_dimension_slice_create(dim->fd.id,
+											  DIMENSION_SLICE_MINVALUE,
+											  DIMENSION_SLICE_MAXVALUE);
+			dimension_slice_insert_relation(rel, slice);
+		}
+		else
+		{
+			Assert(vec->num_slices == 1);
+			slice = vec->slices[0];
+		}
+
+		ts_chunk_constraints_add(chunk->constraints, chunk->fd.id, slice->fd.id, NULL, NULL, true);
+	}
+
+	table_close(rel, RowExclusiveLock);
+
+	return i;
+}
+
+/*
+ * Update correlated constraints dimension slices in the catalog for the
+ * provided chunk (it's assumed that the chunk is locked
+ * appropriately).
+ *
+ * Calculate actual ranges for the given chunk for correlated
+ * constraints and update these entries. This allows for the
+ * chunk to be picked up when queries use correlated columns in
+ * WHERE clauses with these ranges.
+ *
+ * If "reset" is specified then we change the range back to MINVALUE
+ * and MAXVALUE
+ *
+ * Returns the number of slices updated.
+ */
+int
```

Comment:
```suggestion
void
```
The caller is not using the returned value!